### PR TITLE
Compiler 2022 API usage improvement

### PR DESCRIPTION
**Pull request by KulikovNikita**

### CODE REVIEW

File: cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_dense_impl_dpc.cpp [hunk 1]
```diff
@@ -646,9 +646,9 @@ compute_kernel_dense_impl<Float, List>::merge_blocks(local_buffer_list<Float, Li
         local_accessor_rw_t<Float> lmean_buf(local_buffer_size, cgh);
 
         cgh.parallel_for(nd_range, [=](sycl::nd_item<1> item) {
-            const std::int64_t local_size = item.get_local_range()[0];
-            const std::int64_t id = item.get_local_id()[0];
-            const std::int64_t group_id = item.get_group().get_group_id(0);
+            const std::int64_t local_size = item.get_local_range(0);
+            const std::int64_t id = item.get_local_id(0);
+            const std::int64_t group_id = item.get_group(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_dense_impl_dpc.cpp:651]:
```suggestion
            const std::int64_t local_size = item.get_local_range()[0];
            const std::int64_t id = item.get_local_id(0);
            const std::int64_t group_id = item.get_group(0);
```

File: cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_dense_impl_dpc.cpp [hunk 2]
```diff
@@ -935,9 +935,9 @@ compute_kernel_dense_impl<Float, List>::compute_single_pass(const pr::ndarray<Fl
 
     auto last_event = q_.submit([&](sycl::handler& cgh) {
         cgh.parallel_for(nd_range, [=](sycl::nd_item<1> item) {
-            const std::int64_t tid = item.get_local_id()[0];
-            const std::int64_t tnum = item.get_local_range()[0];
-            const std::int64_t gid = item.get_group().get_group_id(0);
+            const std::int64_t tid = item.get_local_id(0);
+            const std::int64_t tnum = item.get_local_range(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_dense_impl_dpc.cpp:939]:
```suggestion
            const std::int64_t tnum = item.get_local_range()[0];
```

File: cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_dense_impl_dpc.cpp [hunk 3]
```diff
@@ -1039,9 +1039,9 @@ compute_kernel_dense_impl<Float, List>::compute_by_blocks(const pr::ndarray<Floa
         ONEDAL_PROFILER_TASK(process_blocks, q_);
         last_event = q_.submit([&](sycl::handler& cgh) {
             cgh.parallel_for(nd_range, [=](sycl::nd_item<1> item) {
-                const std::int64_t tid = item.get_local_id()[0];
-                const std::int64_t tnum = item.get_local_range()[0];
-                const std::int64_t gid = item.get_group().get_group_id(0);
+                const std::int64_t tid = item.get_local_id(0);
+                const std::int64_t tnum = item.get_local_range(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_dense_impl_dpc.cpp:1043]:
```suggestion
                const std::int64_t tnum = item.get_local_range()[0];
```

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_kernel_impl_dpc.cpp [hunk 4]
```diff
@@ -142,13 +142,13 @@ infer_kernel_impl<Float, Index, Task>::predict_by_tree_group_weighted(
             cgh.depends_on(deps);
             cgh.depends_on(last_event);
             cgh.parallel_for(nd_range, [=](sycl::nd_item<2> item) {
-                const Index local_id = item.get_local_id()[0];
-                const Index local_size = item.get_local_range()[0];
+                const Index local_id = item.get_local_id(0);
+                const Index local_size = item.get_local_range(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_kernel_impl_dpc.cpp:146]:
```suggestion
                const Index local_size = item.get_local_range()[0];
```

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_kernel_impl_dpc.cpp [hunk 5]
```diff
@@ -351,10 +351,10 @@ infer_kernel_impl<Float, Index, Task>::reduce_tree_group_response(
             if (sbg.get_group_id() > 0) {
                 return;
             }
-            const Index group_id = item.get_group().get_group_id(0);
+            const Index group_id = item.get_group(0);
             const Index n_groups = item.get_group_range(0);
             const Index sub_group_local_id = sbg.get_local_id();
-            const Index sub_group_size = sbg.get_local_range()[0];
+            const Index sub_group_size = sbg.get_local_range(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_kernel_impl_dpc.cpp:357]:
```suggestion
            const Index sub_group_size = sbg.get_local_range()[0];
```

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_kernel_impl_dpc.cpp [hunk 6]
```diff
@@ -430,10 +430,10 @@ infer_kernel_impl<Float, Index, Task>::determine_winner(const infer_context_t& c
     last_event = queue_.submit([&](sycl::handler& cgh) {
         cgh.depends_on(deps);
         cgh.parallel_for(nd_range, [=](sycl::nd_item<1> item) {
-            const Index local_id = item.get_local_id()[0];
-            const Index local_size = item.get_local_range()[0];
+            const Index local_id = item.get_local_id(0);
+            const Index local_size = item.get_local_range(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_kernel_impl_dpc.cpp:434]:
```suggestion
            const Index local_size = item.get_local_range()[0];
```

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_kernel_hist_impl_dpc.cpp [hunk 7]
```diff
@@ -1738,12 +1738,12 @@ sycl::event train_kernel_hist_impl<Float, Bin, Index, Task>::compute_partial_his
     auto event = queue_.submit([&](sycl::handler& cgh) {
         cgh.depends_on(fill_event);
         cgh.parallel_for(nd_range, [=](sycl::nd_item<2> item) {
-            const Index node_idx = item.get_global_id()[1];
+            const Index node_idx = item.get_global_id(0);
             const Index node_id = node_indices_ptr[node_ind_ofs + node_idx];
-            const Index ftr_grp_idx = item.get_local_id()[0];
-            const Index ftr_grp_size = item.get_local_range()[0];
+            const Index ftr_grp_idx = item.get_local_id(0);
+            const Index ftr_grp_size = item.get_local_range(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_kernel_hist_impl_dpc.cpp:1744]:
```suggestion
            const Index ftr_grp_size = item.get_local_range()[0];
```

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_kernel_impl_dpc.cpp (hunk 6 above)
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_kernel_impl_dpc.cpp]:
```suggestion
            const Index local_size = item.get_local_range()[0];
```

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_feature_type_dpc.cpp [hunk 8]
```diff
@@ -130,16 +130,16 @@ sycl::event indexed_features<Float, Bin, Index>::fill_bin_map(
         cgh.parallel_for(nd_range, [=](sycl::nd_item<1> item) {
             auto sbg = item.get_sub_group();
             const std::uint32_t n_groups = item.get_group_range(0);
-            const std::uint32_t n_sub_groups = sbg.get_group_range()[0];
+            const std::uint32_t n_sub_groups = sbg.get_group_range(0);
             const std::uint32_t n_total_sub_groups = n_sub_groups * n_groups;
             const Index elems_for_sbg =
                 row_count / n_total_sub_groups + bool(row_count % n_total_sub_groups);
-            const std::uint32_t local_size = sbg.get_local_range()[0];
+            const std::uint32_t local_size = sbg.get_local_range(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_feature_type_dpc.cpp:137]:
```suggestion
            const std::uint32_t local_size = sbg.get_local_range()[0];
```

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_kernel_hist_impl_dpc.cpp [hunk 9]
```diff
@@ -1832,12 +1832,12 @@ sycl::event train_kernel_hist_impl<Float, Bin, Index, Task>::compute_partial_cou
     auto event = queue_.submit([&](sycl::handler& cgh) {
         cgh.depends_on(fill_event);
         cgh.parallel_for(nd_range, [=](sycl::nd_item<2> item) {
-            const Index node_idx = item.get_global_id()[1];
+            const Index node_idx = item.get_global_id(1);
             const Index node_id = node_indices_ptr[node_ind_ofs + node_idx];
-            const Index ftr_grp_idx = item.get_local_id()[0];
-            const Index ftr_grp_size = item.get_local_range()[0];
+            const Index ftr_grp_idx = item.get_local_id(0);
+            const Index ftr_grp_size = item.get_local_range(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_kernel_hist_impl_dpc.cpp:1838]:
```suggestion
            const Index ftr_grp_size = item.get_local_range()[0];
```

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_kernel_hist_impl_dpc.cpp [hunk 10]
```diff
@@ -1931,12 +1931,12 @@ sycl::event train_kernel_hist_impl<Float, Bin, Index, Task>::compute_partial_sum
...
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_kernel_hist_impl_dpc.cpp:1937]:
```suggestion
            const Index ftr_grp_size = item.get_local_range()[0];
```

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_service_kernels_dpc.cpp [hunk 11]
```diff
@@ -632,16 +632,16 @@ sycl::event train_service_kernels<Float, Bin, Index, Task>::mark_present_rows(
...
 
             const Index n_groups = item.get_group_range(0);
-            const Index n_sub_groups = sbg.get_group_range()[0];
+            const Index n_sub_groups = sbg.get_group_range(0);
             const Index n_total_sub_groups = n_sub_groups * n_groups;
             const Index elems_for_sbg =
                 node_row_count / n_total_sub_groups + bool(node_row_count % n_total_sub_groups);
 
-            const Index local_size = sbg.get_local_range()[0];
+            const Index local_size = sbg.get_local_range(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_service_kernels_dpc.cpp:640]:
```suggestion
            const Index local_size = sbg.get_local_range()[0];
```

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_service_kernels_dpc.cpp [hunk 12]
```diff
@@ -799,16 +799,16 @@ sycl::event train_service_kernels<Float, Bin, Index, Task>::fill_oob_rows_list_b
...
 
             const Index n_groups = item.get_group_range(0);
-            const Index n_sub_groups = sbg.get_group_range()[0];
+            const Index n_sub_groups = sbg.get_group_range(0);
             const Index n_total_sub_groups = n_sub_groups * n_groups;
             const Index elems_for_sbg =
                 block_row_count / n_total_sub_groups + bool(block_row_count % n_total_sub_groups);
 
-            const Index local_size = sbg.get_local_range()[0];
+            const Index local_size = sbg.get_local_range(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_service_kernels_dpc.cpp:807]:
```suggestion
            const Index local_size = sbg.get_local_range()[0];
```

Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_service_kernels_dpc.cpp]:
```suggestion
            const Index local_size = sbg.get_local_range()[0];
```

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_service_kernels_dpc.cpp [hunk 13]
```diff
@@ -684,16 +684,16 @@ sycl::event train_service_kernels<Float, Bin, Index, Task>::count_absent_rows_fo
...
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_service_kernels_dpc.cpp:692]:
```suggestion
            const Index local_size = sbg.get_local_range()[0];
```

File: cpp/oneapi/dal/backend/primitives/selection/select_flagged_dpc.cpp [hunk 14]
```diff
@@ -45,16 +45,16 @@ sycl::event select_flagged_base<Data, Flag>::scan(sycl::queue& queue,
             auto sbg = item.get_sub_group();
 
             const integer_t n_groups = item.get_group_range(0);
-            const integer_t n_sub_groups = sbg.get_group_range()[0];
+            const integer_t n_sub_groups = sbg.get_group_range(0);
             const integer_t n_total_sub_groups = n_sub_groups * n_groups;
             const integer_t elems_for_sbg =
                 elem_count / n_total_sub_groups + bool(elem_count % n_total_sub_groups);
-            const integer_t local_size = sbg.get_local_range()[0];
+            const integer_t local_size = sbg.get_local_range(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/selection/select_flagged_dpc.cpp:52]:
```suggestion
            const integer_t local_size = sbg.get_local_range()[0];
```

File: cpp/oneapi/dal/backend/primitives/selection/select_flagged_dpc.cpp [hunk 15]
```diff
@@ -147,16 +147,16 @@ sycl::event select_flagged_base<Data, Flag>::reorder(sycl::queue& queue,
...
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/selection/select_flagged_dpc.cpp:154]:
```suggestion
            const integer_t local_size = sbg.get_local_range()[0];
```

File: cpp/oneapi/dal/backend/primitives/sort/sort_dpc.cpp [hunk 16]
```diff
@@ -59,16 +59,16 @@ sycl::event radix_sort_indices_inplace<Float, Index>::radix_scan(sycl::queue& qu
                 return;
             }
             const std::uint32_t n_groups = item.get_group_range(0);
-            const std::uint32_t n_sub_groups = sbg.get_group_range()[0];
+            const std::uint32_t n_sub_groups = sbg.get_group_range(0);
             const std::uint32_t n_total_sub_groups = n_sub_groups * n_groups;
             const Index elems_for_sbg =
                 elem_count / n_total_sub_groups + bool(elem_count % n_total_sub_groups);
-            const std::uint32_t local_size = sbg.get_local_range()[0];
+            const std::uint32_t local_size = sbg.get_local_range(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/sort/sort_dpc.cpp:66]:
```suggestion
            const std::uint32_t local_size = sbg.get_local_range()[0];
```

File: cpp/oneapi/dal/backend/primitives/sort/sort_dpc.cpp [hunk 17]
```diff
@@ -190,16 +190,16 @@ sycl::event radix_sort_indices_inplace<Float, Index>::radix_reorder(
             }
 
             const std::uint32_t n_groups = item.get_group_range(0);
-            const std::uint32_t n_sub_groups = sbg.get_group_range()[0];
+            const std::uint32_t n_sub_groups = sbg.get_group_range(0);
             const std::uint32_t n_total_sub_groups = n_sub_groups * n_groups;
             const Index elems_for_sbg =
                 elem_count / n_total_sub_groups + bool(elem_count % n_total_sub_groups);
-            const std::uint32_t local_size = sbg.get_local_range()[0];
+            const std::uint32_t local_size = sbg.get_local_range(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/backend/primitives/sort/sort_dpc.cpp:197]:
```suggestion
            const std::uint32_t local_size = sbg.get_local_range()[0];
```

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_feature_type_dpc.cpp [hunk 18]
```diff
@@ -130,7 +130,7 @@ sycl::event indexed_features<Float, Bin, Index>::fill_bin_map(
         cgh.parallel_for(nd_range, [=](sycl::nd_item<1> item) {
             auto sbg = item.get_sub_group();
             const std::uint32_t n_groups = item.get_group_range(0);
-            const std::uint32_t n_sub_groups = sbg.get_group_range()[0];
+            const std::uint32_t n_sub_groups = sbg.get_group_range(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_feature_type_dpc.cpp:133]:
```suggestion
            const std::uint32_t n_sub_groups = sbg.get_group_range()[0];
```

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_service_kernels_dpc.cpp [hunk 19]
```diff
@@ -632,7 +632,7 @@ sycl::event train_service_kernels<Float, Bin, Index, Task>::mark_present_rows(
             auto sbg = item.get_sub_group();
 
             const Index n_groups = item.get_group_range(0);
-            const Index n_sub_groups = sbg.get_group_range()[0];
+            const Index n_sub_groups = sbg.get_group_range(0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_service_kernels_dpc.cpp:635]:
```suggestion
            const Index n_sub_groups = sbg.get_group_range()[0];
```

**Review by makart19 — APPROVED** [review]

**Comment by napetrov** [comment]
@KulikovNikita  - does it make sense to merge this?

**Pull request closed without merging**